### PR TITLE
feat(modules): add artemis container

### PR DESCRIPTION
**Pull request by abemedia**
## What does this PR do?

Adds Apache ActiveMQ Artemis module.

## Why is it important?

Artemis supports most common messaging protocols (AMQP, MQTT, STOMP) so this allows a fairly wide range of testing integrations with message broker clients.

## Related issues

-

### CODE REVIEW

File: modules/artemis/example_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package artemis_test
+
```
Comment (mdelapenya) [inline modules/artemis/example_test.go:2]: Oh super neat to not embed test code into the docs 👏👏👏

**Review by mdelapenya — COMMENTED** [review]
Thanks for creating this module, at first sight it looks great!

I'm interested in your use case for this technology. Nevertheless, I will review it more deeply next Monday.

**Comment by abemedia** [comment]
I'm currently working on a library similar to the standard library's `database/sql` but for message broker clients and this container lets me test three of the drivers.

File: modules/artemis/example_test.go (hunk 1 above)
Comment (abemedia) [reply modules/artemis/example_test.go:2]: 🙂

Comment (mdelapenya) [reply modules/artemis/example_test.go:2]: BTW I'm planning to have an example generated alongside the module, which would simplify any code snippet. Thanks for the idea! 💡

File: modules/artemis/artemis_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+				t.Fatal(err)
+			}
+			t.Cleanup(func() {
+				if err := container.Terminate(ctx); err != nil {
+					t.Fatalf("failed to terminate container: %s", err)
+				}
+			})
+
+			u, err := container.ConsoleURL(ctx)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (abemedia) [reply modules/artemis/artemis_test.go:69]: Sure. I personally find standard Go tests much easier to read than `testify` but your repo, your rules 😜 
Will update the PR tomorrow.

**Review by eddumelendez — COMMENTED** [review]
Thanks for your contribution! I've been waiting for artemis and activemq images and thanks to your PR I realized those were available since couple of weeks ago.

File: modules/artemis/artemis.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+// Additional arguments sent to the `artemis create“ command.
+// The default is `--http-host 0.0.0.0 --relax-jolokia`.
+// Setting this value will override the default.
+// See the documentation on `artemis create` for available options.
+func WithExtraArgs(args string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["EXTRA_ARGS"] = args
+	}
+}
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/artemis/artemis.go:72]: I have a personal preference on adding a little bit of sugar on certain methods that are very useful for the consumer of the module.

For this particular case, as a non-user of Artemis. I have no opinion if passing the EXTRA_ARGS as part of the `req.Env` or in this specialised functional option is fine.

@abemedia what are your thoughts on this?

File: modules/artemis/example_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	conn, err := stomp.Dial("tcp", host, stomp.ConnOpt.Login(user, pass))
+	if err != nil {
+		panic(err)
+	}
+
+	_ = conn
```
Comment (mdelapenya) [inline modules/artemis/example_test.go:33]: Do we want to terminate the container here in the example to make the code snippet self-explanatory?

File: modules/artemis/artemis.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (abemedia) [reply modules/artemis/artemis.go:72]: I personally feel it's still useful to have these, just for convenience. Happy to remove if you guys disagree.

File: docs/modules/artemis.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+#### Docker type modifiers
+
+If you need an advanced configuration for Artemis, you can leverage the following Docker type modifiers:
+
+- `testcontainers.WithConfigModifier`
+- `testcontainers.WithHostConfigModifier`
+- `testcontainers.WithEndpointSettingsModifier`
+
+Please read the [Create containers: Advanced Settings](../features/creating_container.md#advanced-settings) documentation for more information.
+
```
[the remark this answers is not in the document]
Comment (abemedia) [reply docs/modules/artemis.md:89]: Done

File: docs/modules/artemis.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+### Container Methods
+
+The Artemis container exposes the following methods:
```
[the remark this answers is not in the document]
Comment (abemedia) [reply docs/modules/artemis.md:92]: Done

**Comment by abemedia** [comment]
Hey @mdelapenya @eddumelendez, sorry about the delay but I was ill during the week.
Have just addressed all the PR comments and rebased on main.

File: modules/artemis/example_test.go (hunk 4 above)
Comment (abemedia) [reply modules/artemis/example_test.go]: Done

**Review by mdelapenya — APPROVED** [review]
I think we are good to go! As mentioned in the new comments, I'll add the code snippets in a follow-up PR. 🚀

Thank you very much for your patience while reviewing the module  🙇

**Comment by mdelapenya** [comment]
Merge now, as the CI build already passed and the latest commit updates go.mod, only

**Pull request merged**